main.tf

```
@@ -1,3 +1,40 @@
+locals {
+  egress_rules = merge([
+    for rule_key, rule in var.egress_rules : {
+      for cidr in concat(
+        rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [],
+        rule.cidr_ipv6 != null ? rule.cidr_ipv6 : []
+        ) : "${rule_key}-${cidr}" => {
+        description                  = rule.description
+        from_port                    = rule.from_port
+        ip_protocol                  = rule.ip_protocol
+        to_port                      = rule.to_port
+        cidr_ipv4                    = contains(rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [], cidr) ? cidr : null
+        cidr_ipv6                    = contains(rule.cidr_ipv6 != null ? rule.cidr_ipv6 : [], cidr) ? cidr : null
+        prefix_list_id               = rule.prefix_list_id
+        referenced_security_group_id = rule.referenced_security_group_id
+      }
+    }
+  ]...)
+  ingress_rules = merge([
+    for rule_key, rule in var.ingress_rules : {
+      for cidr in concat(
+        rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [],
+        rule.cidr_ipv6 != null ? rule.cidr_ipv6 : []
+        ) : "${rule_key}-${cidr}" => {
+        description                  = rule.description
+        from_port                    = rule.from_port
+        ip_protocol                  = rule.ip_protocol
+        to_port                      = rule.to_port
+        cidr_ipv4                    = contains(rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [], cidr) ? cidr : null
+        cidr_ipv6                    = contains(rule.cidr_ipv6 != null ? rule.cidr_ipv6 : [], cidr) ? cidr : null
+        prefix_list_id               = rule.prefix_list_id
+        referenced_security_group_id = rule.referenced_security_group_id
+      }
+    }
+  ]...)
```

This could be more concise:

```suggestion
  egress_rules = merge([
    for rule_key, rule in var.egress_rules : {
      for cidr in concat(
        rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [],
        rule.cidr_ipv6 != null ? rule.cidr_ipv6 : []
        ) : "${rule_key}-${cidr}" => merge(rule, {
          cidr_ipv4 = contains(rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [], cidr) ? cidr : null
          cidr_ipv6 = contains(rule.cidr_ipv6 != null ? rule.cidr_ipv6 : [], cidr) ? cidr : null
      })
    }
  ]...)
  ingress_rules = merge([
    for rule_key, rule in var.ingress_rules : {
      for cidr in concat(
        rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [],
        rule.cidr_ipv6 != null ? rule.cidr_ipv6 : []
        ) : "${rule_key}-${cidr}" => merge(rule, {
          cidr_ipv4 = contains(rule.cidr_ipv4 != null ? rule.cidr_ipv4 : [], cidr) ? cidr : null
          cidr_ipv6 = contains(rule.cidr_ipv6 != null ? rule.cidr_ipv6 : [], cidr) ? cidr : null
      })
    }
  ]...)
```